components/testmonitor/actions/common/base.mjs

```diff
@@ -0,0 +1,102 @@
+import testmonitor from "../../testmonitor.app.mjs";
+
+export default {
+  props: {
+    testmonitor,
+    limit: {
+      propDefinition: [
+        testmonitor,
+        "limit",
+      ],
+      optional: true,
+    },
+    page: {
+      propDefinition: [
+        testmonitor,
+        "page",
+      ],
+      optional: true,
+    },
```

Comment:
Rather than `limit` and `page` being props, the actions should paginate through the results. I'd suggest adding a `max` prop where the user can set the maximum number of results to return, and then the action should paginate through and return a maximum of `max` results.